js/battle.js

```diff
@@ -4129,16 +4139,24 @@ var Battle = (function () {
 					actions += "" + poke.getName() + " transformed!";
 					if (toId(template.species) === 'shaymin') break;
 				} else if (toId(template.species) === 'darmanitanzen') {
```

Comment:
@ascriptmaster, I finally managed to test it. Zen Mode ended after the Skill Swap, so yeah, it's like Flash Fire / Unburden.